src/credentials/GroupDataProvider.h

```diff
@@ -138,6 +146,21 @@ class GroupDataProvider
             VerifyOrReturnError(this->policy == other.policy && this->num_keys_used == other.num_keys_used, false);
             return !memcmp(this->epoch_keys, other.epoch_keys, this->num_keys_used * sizeof(EpochKey));
         }
+
+        ByteSpan GetCurrentKey()
+        {
+            if (this->num_keys_used > 0)
+            {
+                // An epoch key update SHALL order the keys from oldest to newest,
+                // the current epoch key having the second newest time
+                size_t current_index = (this->num_keys_used > 2) ? 1 : 0;
+                return ByteSpan(epoch_keys[current_index].key, EpochKey::kLengthBytes);
+            }
+            else
+            {
+                return ByteSpan(nullptr, 0);
+            }
```

Comment:
There are a total of four possible cases here:

* 0 keys: return ByteSpan(nullptr, 0)
* 1 key: return index 0
* 2 keys: return index 0
* 3 keys: return index 1

Yet, this is implemented as a ternary nested within an if-else.  This ins't very readable.  Why can't the code present the four possible cases in an obvious way?

Reply:
I am afraid you're right, this isn't pretty. I have replace this with a switch, which I believe is a lot more readable.